pkg/network/certificates_knobs.go

```diff
@@ -0,0 +1,99 @@
+package network
+
+import (
+	"reflect"
+	"time"
+
+	"github.com/pkg/errors"
+
+	opv1alpha1 "github.com/kubevirt/cluster-network-addons-operator/pkg/apis/networkaddonsoperator/v1alpha1"
+)
+
+// validateCertificateKnobs validates the following fields
+// - CARotateInterval
+// - CAOverlapInterval
+// - CertRotateInterval
+func validateCertificateKnobs(conf *opv1alpha1.NetworkAddonsConfigSpec) []error {
+	if conf.CertificateKnobs == nil {
+		return []error{}
+	}
+
+	certificateKnobs := *conf.CertificateKnobs
+
+	errs := []error{}
+
+	caRotateInterval := parseCertificateKnob("caRotateInterval", certificateKnobs.CARotateInterval, &errs)
+	caOverlapInterval := parseCertificateKnob("caOverlapInterval", certificateKnobs.CAOverlapInterval, &errs)
+	certRotateInterval := parseCertificateKnob("certRotateInterval", certificateKnobs.CertRotateInterval, &errs)
+
+	// If they cannot be parsed don't continue
+	if len(errs) > 0 {
+		return errs
+	}
+
+	validateGreaterThanZero("caRotateInterval", caRotateInterval, &errs)
+	validateGreaterThanZero("caOverlapInterval", caOverlapInterval, &errs)
+	validateGreaterThanZero("certRotateInterval", certRotateInterval, &errs)
+
+	// If we have a zero value don't continue
+	if len(errs) > 0 {
+		return errs
+	}
+
+	validateGreaterThan("caRotateInterval", caRotateInterval, "caOverlapInterval", caOverlapInterval, &errs)
+	validateGreaterThan("caRotateInterval", caRotateInterval, "certRotateInterval", certRotateInterval, &errs)
+
+	return errs
+}
+
+func fillDefaultsCertificateKnobs(conf, previous *opv1alpha1.NetworkAddonsConfigSpec) []error {
+	if conf.CertificateKnobs == nil {
+		return []error{}
+	}
+
+	if conf.CertificateKnobs.CARotateInterval == "" || conf.CertificateKnobs.CAOverlapInterval == "" || conf.CertificateKnobs.CertRotateInterval == "" {
```

Comment:
When would this happen? changeSafeCertificateKnobs checks against this option, doesn't it?

Reply:
Aha, nevermind, it will be useful to bring the previously implicitly configured values. Just like it is done in https://github.com/kubevirt/cluster-network-addons-operator/blob/36f9b55ac1a1f6481960fbca84fe50dcc58b235e/pkg/network/kubemacpool.go#L66

Reply:
If some of the knobs is missing we set to defaults all of them, since we don't want to revalidate here, is kind of all or nothing.